internal/store/store.go

```diff
@@ -484,35 +489,110 @@ func (s *Store) PushCompose(composeID uuid.UUID, bp *blueprint.Blueprint, checks
 		repos = append(repos, source.RepoConfig())
 	}
 
-	size = s.distro.GetSizeForOutputType(composeType, size)
-
-	pipeline, err := s.distro.Pipeline(bp, repos, checksums, arch, composeType, size)
+	pipelineStruct, err := s.distro.Pipeline(bp, repos, checksums, arch, composeType, size)
 	if err != nil {
 		return err
 	}
-	s.change(func() error {
-		s.Composes[composeID] = Compose{
-			QueueStatus: "WAITING",
-			Blueprint:   bp,
-			Pipeline:    pipeline,
-			OutputType:  composeType,
-			Targets:     targets,
-			JobCreated:  time.Now(),
-			Size:        size,
+	// FIXME: handle or comment this possible error
+	_ = s.change(func() error {
+		s.Composes[composeID] = compose.Compose{
+			Blueprint: bp,
+			ImageBuilds: []compose.ImageBuild{
+				{
+					QueueStatus: common.IBWaiting,
```

Comment:
Why is there no `Distro` field? I think it might confuse us in the future.

Reply:
Because this is the compatibility version, which is supposed to replace previous pushcompose without any changes.

Reply:
Makes sense, we can clean-up in the follow-up.